### ISSUE TITLE
Dashboard does not load settings from current config

### ISSUE THREAD

**Issue by richard-scott** [emqx/emqx#9201]
### What happened?

I am unable to see settings previously updated via the Dashboard, when I go to see settings for the MQTT Configuration I have to re-enter ALL missing settings every single time I want to change one setting otherwise I get errors like this:

<img width="1475" alt="Screenshot 2022-10-21 at 10 01 55" src="https://user-images.githubusercontent.com/5149834/197160547-ad4069e0-f72d-4bcc-85bc-c2505975e86d.png">

### What did you expect to happen?

I want the settings to re-load in the Dashboard when using it.

### How can we reproduce it (as minimally and precisely as possible)?

In the Dashboard, go into Configuration -> MQTT and change any blank setting to something. Click Save, and you'll get the error about missing items. Complete the missing items and click save. This should save.

Now go back to the Dashboard home, and then go back into the Configuration -> MQTT section and you'll see that nothing has been loaded from the previously saved edit.

### EMQX version

<details>

```console
$ ./bin/emqx_ctl broker
sysdescr  : EMQX
version   : 5.0.8
datetime  : 2022-10-17T20:05:56.393539406+00:00
uptime    : 6 hours, 55 minutes, 33 seconds
```

</details>

### OS version

<details>

```console
# On Linux:
$ cat /etc/os-release
PRETTY_NAME="Debian GNU/Linux 11 (bullseye)"
NAME="Debian GNU/Linux"
VERSION_ID="11"
VERSION="11 (bullseye)"
VERSION_CODENAME=bullseye
ID=debian
HOME_URL="https://www.debian.org/"
SUPPORT_URL="https://www.debian.org/support"
BUG_REPORT_URL="https://bugs.debian.org/"
$ uname -a
Linux ip-10-254-74-172 5.10.0-18-cloud-arm64 #1 SMP Debian 5.10.140-1 (2022-09-02) aarch64 GNU/Linux
```

</details>

### Log files

<details>

</details>

**Comment by richard-scott** [comment]
I've also noticed that the 'defaults' described in the [Docs](https://www.emqx.io/docs/en/v5.0/admin/cfg.html#broker-mqtt), are not represented in the Dashboard.

One's such as @path mqtt.retain\_available defaulting to true in the Docs, shows up as disabled in the Dashboard.

**Comment by richard-scott** [comment]
However, if you set the default value to the correct one in /etc/emqx/emqx.conf then the Dashboard displays it.

**Comment by zhongwencool** [comment]
@richard-scott  sorry for this delay.
I can't reproduce this error on v5.0.8 with default emqx.conf, could you provide `emqx.log.1` and `etc/emqx.conf`.

If only according to the above error message, it should be dashboard front-end did not pass the corresponding value to emqx server. 
To better find the root cause, open the developer mode of your browser and post the entire request/response.

**Comment by richard-scott** [comment]
It is still the same with 5.0.10. Install a fresh system and in the Dashboard click on Configuration, then click on MQTT, then click Save.

**Comment by zhongwencool** [comment]
<img width="1459" alt="image" src="https://user-images.githubusercontent.com/3116225/202060526-d6121a1f-7ec1-401a-b86c-c41a87c1d542.png">

I installed 5.0.10 and opened the MQTT tab, it's normal. :(
Are there any additional steps I'm not aware of?
Please help find more log: **open the developer mode of your browser and post the entire request/response.**

> It is still the same with 5.0.10. Install a fresh system and in the Dashboard click on Configuration, then click on MQTT, then click Save.

**Comment by richard-scott** [comment]
Request:

```
{"conn_congestion":{"enable_alarm":true,"min_alarm_sustain_duration":"1m"},
"flapping_detect":{"ban_time":"5m","enable":false,"max_count":15,"window_time":"1m"},
"force_gc":{"bytes":"16MB","count":16000,"enable":true},
"force_shutdown":{"enable":true,"max_heap_size":"32MB","max_message_queue_len":1000},
"mqtt":{"await_rel_timeout":"300s","exclusive_subscription":false,"idle_timeout":"1200s",
"ignore_loop_deliver":false,"keepalive_backoff":0.75,"max_awaiting_rel":100,"max_clientid_len":128,
"max_inflight":32,"max_packet_size":"128KB","max_qos_allowed":2,"max_subscriptions":"infinity",
"max_topic_alias":0,"max_topic_levels":16,"peer_cert_as_clientid":"disabled",
"peer_cert_as_username":"disabled","response_information":"","retain_available":true,
"retry_interval":30,"server_keepalive":30,"session_expiry_interval":"168h",
"shared_subscription":true,"strict_mode":true,"upgrade_qos":false,"use_username_as_clientid":false,
"wildcard_subscription":true},"overload_protection":{"enable":true},"stats":{"enable":true}}
```

Response:  
`{"code":"BAD_REQUEST","message":"[{\"kind\":\"validation_error\",\"mismatches\":{\"infinity\":{\"kind\":\"validation_error\",\"path\":\"mqtt.max_mqueue_len\",\"reason\":\"mandatory_required_field\",\"value\":\"undefined\"},\"non_neg_integer()\":{\"kind\":\"validation_error\",\"path\":\"mqtt.max_mqueue_len\",\"reason\":\"mandatory_required_field\",\"value\":\"undefined\"}},\"path\":\"mqtt.max_mqueue_len\",\"reason\":\"matched_no_union_member\"},{\"kind\":\"validation_error\",\"mismatches\":{\"disabled\":{\"kind\":\"validation_error\",\"path\":\"mqtt.mqueue_priorities\",\"reason\":\"mandatory_required_field\",\"value\":\"undefined\"},\"map()\":{\"kind\":\"validation_error\",\"path\":\"mqtt.mqueue_priorities\",\"reason\":\"mandatory_required_field\",\"value\":\"undefined\"}},\"path\":\"mqtt.mqueue_priorities\",\"reason\":\"matched_no_union_member\"},{\"kind\":\"validation_error\",\"path\":\"mqtt.mqueue_default_priority\",\"reason\":\"mandatory_required_field\",\"value\":\"undefined\"},{\"kind\":\"validation_error\",\"path\":\"mqtt.mqueue_store_qos0\",\"reason\":\"mandatory_required_field\",\"value\":\"undefined\"}]"}`

**Comment by richard-scott** [comment]
In your screenshot, you have different settings from the defaults on a fresh install. Please try on a brand new setup with no previous config around.

**Comment by zhongwencool** [comment]
> In your screenshot, you have different settings from the defaults on a fresh install. Please try on a brand new setup with no previous config around.

I don't think so. 

In the init config, each item has a default value, but in your request, some MQTT items are missing. It looks like the full list of configuration items is not available when the page is opened. 
1. what's the response of  GET `/api/v5/configs/global_zone` ?
2. `./bin/emqx eval "emqx:get_raw_config([mqtt])."`
3. `./bin/emqx eval "emqx:get_config([mqtt])."`

**Comment by richard-scott** [comment]
global\_zone:

```
{
  "conn_congestion":{
     "enable_alarm":true,
     "min_alarm_sustain_duration":"1m"
  },
  "flapping_detect":{
     "ban_time":"5m",
     "enable":false,
     "max_count":15,
     "window_time":"1m"
  },
  "force_gc":{
     "bytes":"16MB",
     "count":16000,
     "enable":true
  },
  "force_shutdown":{
     "enable":true,
     "max_heap_size":"32MB",
     "max_message_queue_len":1000
  },
  "mqtt":{
     "await_rel_timeout":"300s",
     "exclusive_subscription":false,
     "idle_timeout":"1200s",
     "ignore_loop_deliver":false,
     "keepalive_backoff":0.75,
     "max_awaiting_rel":100,
     "max_clientid_len":128,
     "max_inflight":32,
     "max_packet_size":"128KB",
     "max_qos_allowed":2,
     "max_subscriptions":"infinity",
     "max_topic_alias":0,
     "max_topic_levels":16,
     "peer_cert_as_clientid":"disabled",
     "peer_cert_as_username":"disabled",
     "response_information":"",
     "retain_available":true,
     "retry_interval":30,
     "server_keepalive":30,
     "session_expiry_interval":"168h",
     "shared_subscription":true,
     "strict_mode":true,
     "upgrade_qos":false,
     "use_username_as_clientid":false,
     "wildcard_subscription":true
  },
  "overload_protection":{
     "enable":true
  },
  "stats":{
     "enable":true
  }
}
```

```
# emqx eval "emqx:get_raw_config([mqtt])."
#{<<"await_rel_timeout">> => <<"300s">>,<<"exclusive_subscription">> => false,
 <<"idle_timeout">> => <<"1200s">>,<<"ignore_loop_deliver">> => false,
 <<"keepalive_backoff">> => 0.75,<<"max_awaiting_rel">> => 100,
 <<"max_clientid_len">> => 128,<<"max_inflight">> => 32,
 <<"max_packet_size">> => <<"128KB">>,<<"max_qos_allowed">> => 2,
 <<"max_subscriptions">> => <<"infinity">>,<<"max_topic_alias">> => 0,
 <<"max_topic_levels">> => 16,<<"peer_cert_as_clientid">> => <<"disabled">>,
 <<"peer_cert_as_username">> => <<"disabled">>,
 <<"response_information">> => <<>>,<<"retain_available">> => true,
 <<"retry_interval">> => 30,<<"server_keepalive">> => 30,
 <<"session_expiry_interval">> => <<"168h">>,
 <<"shared_subscription">> => true,<<"strict_mode">> => true,
 <<"upgrade_qos">> => false,<<"use_username_as_clientid">> => false,
 <<"wildcard_subscription">> => true}
```

```
# emqx eval "emqx:get_config([mqtt])."
#{await_rel_timeout => 300000,exclusive_subscription => false,
 idle_timeout => 1200000,ignore_loop_deliver => false,
 keepalive_backoff => 0.75,max_awaiting_rel => 100,max_clientid_len => 128,
 max_inflight => 32,max_mqueue_len => 1000,max_packet_size => 131072,
 max_qos_allowed => 2,max_subscriptions => infinity,max_topic_alias => 0,
 max_topic_levels => 16,mqueue_default_priority => lowest,
 mqueue_priorities => disabled,mqueue_store_qos0 => true,
 peer_cert_as_clientid => disabled,peer_cert_as_username => disabled,
 response_information => [],retain_available => true,retry_interval => 30,
 server_keepalive => 30,session_expiry_interval => 604800000,
 shared_subscription => true,strict_mode => true,upgrade_qos => false,
 use_username_as_clientid => false,wildcard_subscription => true}
```

**Comment by zhongwencool** [comment]
Thank you very much for such detailed logs.  

we will fix this bug next version `5.0.11`, But you can avoid this error by doing the following before the release of a new version.
1. Remove `mqtt {}` from `emqx.conf`.
2. Delete `data/configs/cluster-override.conf`
3. Restart EMQX
4. Modify the mqtt configuration via the dashboard only.

**Comment by richard-scott** [comment]
Thanks for the update.

I only put the items into `mqtt {}` in `emqx.conf` as without this, the defaults were not showing up in the Dashboard.
